src/utils/generateJailsChartData.js

```diff
@@ -0,0 +1,127 @@
+// Recidiviz - a data platform for criminal justice reform
+// Copyright (C) 2020 Recidiviz, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+// =============================================================================
+import { COUNTIES_NOT_PROVIDED } from "../constants/errors";
+import formatNumber from "./formatNumber";
+import logger from "./logger";
+
+export const noMetricData = (metric) =>
+  `${metric} doesn't appear in source files. Chart data is not generated.`;
+
+/**
+ * Transforms normalized data to chart.js format.
+ * @param data - normalized data (see @returns of getNormalizedStateData)
+ * @param metric - metric name for which we generate chart data
+ * @param counties {string[]} - counties for which we generate chart data
+ * @param countyLabels {string[]} - humanized county names
+ * @param coveredCounty - counties reporting percent
+ * @returns {{
+ * datasets: {
+ *   metric: string
+ *   county: string
+ *   coveredCounty: string
+ *   isNotAvailable: boolean
+ *   isStatewide: boolean
+ *   label: string
+ *   data: (number|null)[]
+ * }[],
+ * labels: string[],
+ * }}
+ */
+
+const generateJailsChartData = (data, metric, counties, countyLabels = [], coveredCounty) => {
```

Comment:
Let's make sure to add unit tests for this one before we merge.

Reply:
Also, the logic in this function looks _very similar_ to the logic in `generateCorrectionsChartData`. Let's try to abstract the common parts out into a helper function? At the very least, it looks like the `periods` creation is identical, unless I'm missing something.

Reply:
Moved the logic of `periods` into another func. The rest is a little bit different in mentioned functions. Tests will be added closer to the end of PR.